tests/conftest.py

```diff
@@ -0,0 +1,125 @@
+from jinja2 import Environment
+from logging import basicConfig, getLogger, INFO
+from OpenSSL import crypto
+from os import path
+import pytest
+from subprocess import (
+    check_output,
+    Popen,
+    STDOUT,
+    SubprocessError,
+    TimeoutExpired,
+)
+from time import sleep
+
+
+pytest_plugins = [
+    "mock_fixtures",
+]
+
+
+def pytest_addoption(parser):
+    parser.addoption("--nginx", default="../nginx/objs/nginx")
+    parser.addoption("--module", default="build/ngx_otel_module.so")
+    parser.addoption("--showlog", action="store_true")
+    parser.addoption("--otelcol")
+
+
+def self_signed_cert(test_dir, name):
+    k = crypto.PKey()
+    k.generate_key(crypto.TYPE_RSA, 2048)
+    cert = crypto.X509()
+    cert.get_subject().CN = name
+    cert.set_issuer(cert.get_subject())
+    cert.gmtime_adj_notBefore(0)
+    cert.gmtime_adj_notAfter(365 * 86400)  # 365 days
+    cert.set_pubkey(k)
+    cert.sign(k, "sha512")
+    (test_dir / f"{name}.key").write_text(
+        crypto.dump_privatekey(crypto.FILETYPE_PEM, k).decode("utf-8")
+    )
+    (test_dir / f"{name}.crt").write_text(
+        crypto.dump_certificate(crypto.FILETYPE_PEM, cert).decode("utf-8")
+    )
+
+
+@pytest.fixture(scope="session")
+def logger():
+    basicConfig(level=INFO)
+    return getLogger(__name__)
+
+
+@pytest.fixture(scope="module")
+def testdir(tmp_path_factory):
+    return tmp_path_factory.mktemp("nginx")
+
+
+@pytest.fixture(scope="module", autouse=True)
+def _errorlog(pytestconfig, logger, testdir):
+    yield
+    if pytestconfig.option.showlog:
+        if (testdir / "error.log").exists():
+            logger.debug((testdir / "error.log").read_text())
```

Comment:
This should be printed regardless of log level.

Reply:
Done.